### PR TITLE
Refactor replace.js - move php code into one place

### PR DESCRIPTION
**Pull request by daisy1754**
# Description
Move all php code in replacejs.phtml into top of file so in subsequent PR I can move js into separate file

Fixes: (link Asana Task)
https://app.asana.com/0/941920570700290/1147770190679403/f

# Type of change

Refactoring. It changes scope of variables a bit but shouldn't change any functionality.

# How Has This Been Tested?
- [ ] Successfully tested locally (or docker image)
- [ x ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [x] I have added my Asana task link and provided a changelog message if applicable.

### CODE REVIEW

File: view/frontend/templates/js/replacejs.phtml [hunk 1]
```diff
@@ -35,6 +35,62 @@ ob_start();
...
+            } catch (error) {
+                console.error(error);
+            }
+        }
+    };
+    var defaultErrorMessage = "<?php echo $block->getBoltPopupErrorMessage() ?>";
+    var successCallback = function(data) {
+        <?php echo $block->getJavascriptSuccess(); ?>
+    };
+
```
Comment (danac-gs) [inline view/frontend/templates/js/replacejs.phtml]: Moving all the variables above into the global scope might eventually collide with some other piece of code / module. They have pretty common names. Please consider adding them a name prefix or make them properties of some global object (e.g. "Bolt", "Boltpay").

File: view/frontend/templates/js/replacejs.phtml [hunk 2]
```diff
@@ -35,6 +35,62 @@ ob_start();
     // in payment method renderer, vendor/boltpay/bolt-magento2/view/frontend/web/js/view/payment/method-renderer/boltpay.js
     window.boltConfig = <?php echo $block->getSettings(); ?>;
 
+    var trackCallbacks = {
```
Comment (roopakv) [inline view/frontend/templates/js/replacejs.phtml]: question: while you are at it wdyt about creating a php function to output the following js so we dont have that code all repeated. The php fn takes `$trackCallbacks['checkout_start']` as input

```
function() {
  try {
    ...callback_here...
  } catch (error) {
    console.error(error);
  }
}
```

if you have a different plan for future PRs ignore this

File: view/frontend/templates/js/replacejs.phtml [hunk 3]
```diff
@@ -35,6 +35,62 @@ ob_start();
...
+        onCheckoutStart: function() {
+            try {
+                <?php echo $trackCallbacks['checkout_start']; ?>
```
Comment (ethanwayda22) [inline view/frontend/templates/js/replacejs.phtml]: Question: Is this a separate function also called trackCallbacks?

Comment (daisy1754) [reply view/frontend/templates/js/replacejs.phtml]: tweaked a code bit. this is PHP variable with which we construct JS variable named trackCallbacks here

File: view/frontend/templates/js/replacejs.phtml (hunk 2 above)
Comment (daisy1754) [reply view/frontend/templates/js/replacejs.phtml]: done

File: view/frontend/templates/js/replacejs.phtml (hunk 1 above)
Comment (daisy1754) [reply view/frontend/templates/js/replacejs.phtml]: done

File: view/frontend/templates/js/replacejs.phtml [hunk 4]
```diff
@@ -24,7 +24,19 @@
  */
 if ($block->shouldDisableBoltCheckout()) return;
 
-$trackCallbacks = $block->getTrackCallbacks();
+$trackCallbackCode = $block->getTrackCallbacks();
+
+function wrapWithCatch($jsCode, $argName='') {
+echo <<<JavaScript
```
Comment (roopakv) [inline view/frontend/templates/js/replacejs.phtml]: NICE! did not know i could do this :)

**Pull request merged**